.github/workflows/build_and_test.yml

```diff
@@ -251,13 +251,73 @@ jobs:
         name: unit-tests-log-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
         path: "**/target/unit-tests.log"
 
-  pyspark:
+  infra-image:
     needs: precondition
-    if: fromJson(needs.precondition.outputs.required).pyspark == 'true'
+    # Currently, only enable docker build from cache for `master` branch jobs
+    if: fromJson(needs.precondition.outputs.required).pyspark == 'true' && ${{ inputs.branch }} == 'master'
+    runs-on: ubuntu-latest
+    outputs:
+      image_url: ${{ steps.infra-image-outputs.outputs.image_url }}
+    steps:
+      - name: Generate image name and url
+        id: infra-image-outputs
+        run: |
+          # Convert to lowercase to meet docker repo name requirement
+          REPO_OWNER=$(echo "${{ github.repository_owner }}" | tr '[:upper:]' '[:lower:]')
+          IMG_NAME="apache-spark-ci-image:${{ inputs.branch }}-${{ github.run_id }}"
+          IMG_URL="ghcr.io/$REPO_OWNER/$IMG_NAME"
+          echo ::set-output name=image_url::$IMG_URL
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
```

Comment:
TODO: need a test for schedule job before ready to go

Reply:
`github.repository_owner`: `apache` in here according to [here](https://github.com/search?q=org%3Aapache+%22github.repository_owner%22&type=code)
`github.actor`: The latest user who create the schedule trigger will be the actor for the schedule workflow. `Yikun` in here

So, this may bring some issue on apace/spark job (such as schedule, push to master/branches). We can use the `image-cache` directly for jobs under apache/spark. 

[1] https://github.community/t/who-will-be-the-github-actor-when-a-workflow-runs-on-a-schedule/17369

Reply:
Case 1: The developer push a commit: ==> OK 
- github.actor: `yikun`
- generate and use image like `yikun/apache-spark-ci-image:master-xxx`

Case 2: The apache/spark merge a commit in master: ==> OK
- github.actor: `HyukjinKwon`
- generate and use image like `hyukjinkwon/apache-spark-ci-image:master-xxx`

Case 3: The scheduler jobs for master job:  ==> ???
- Use master image cache directly (`spark/apache-spark-github-action-image-cache:master`)

Case 4: The scheduler jobs for branches job  ==> OK
- Use branches image cache directly (`dongjoon/apache-spark-github-action-image:20220207`, future is `spark/apache-spark-github-action-image-cache:branches-*`)

Case 5: The developer push a commit in branches: ==> OK
Case 6: The apache/spark merge a commit in branches:
- use branches workflow, no change by this pr